rust/datafusion/src/physical_plan/parquet.rs

```diff
@@ -611,7 +611,7 @@ fn build_predicate_expression(
             let max_column_expr = expr_builder.max_column_expr()?;
             min_column_expr
                 .lt_eq(expr_builder.scalar_expr().clone())
-                .and(expr_builder.scalar_expr().lt_eq(max_column_expr))
+                .and(expr_builder.scalar_expr().clone().lt_eq(max_column_expr))
```

Comment:
this is the only change where the caller had to do an extra clone (and in this case it was actually needed as `scalar_expr()` is used twice. In other locations the expr was ignored and this can be consumed without issue